tests/api/test_product.py

```diff
@@ -1068,6 +1069,55 @@ def test_update_product(
     assert not data["product"]["category"]["name"] == category.name
 
 
+SET_ATTRIBUTES_TO_PRODUCT_QUERY = """
+    mutation updateProduct($productId: ID!, $attributes: [AttributeValueInput!]) {
+      productUpdate(id: $productId, input: { attributes: $attributes }) {
+        errors {
+          message
+          field
+        }
+      }
+    }
+"""
+
+
+def test_update_product_can_only_assign_multiple_values_to_valid_input_types(
+    staff_api_client, product, permission_manage_products, color_attribute
+):
+    """Ensures you cannot assign multiple values to input types
+    that are not multi-select. This also ensures multi-select types
+    can be assigned multiple values as intended."""
+
+    staff_api_client.user.user_permissions.add(permission_manage_products)
```

Comment:
Why not simply:
```
response = staff_api_client.post_graphql(
        query, variables, permissions=[permission_manage_products]
    )
```
?

Reply:
Because it will do two query which is slowing down tests. There is already a test that is ensuring the permissions are respected for the mutation, so there is no need of verifying N more times.